code/modules/cm_marines/equipment/mortar/mortars.dm

```diff
@@ -209,37 +211,44 @@
 /obj/structure/mortar/attackby(obj/item/O, mob/user)
 	if(istype(O, /obj/item/mortar_shell))
 		var/obj/item/mortar_shell/mortar_shell = O
+		var/turf/T = locate(targ_x + dial_x + offset_x, targ_y + dial_y + offset_y, z)
+		var/area/A = get_area(T)
 		if(!skillcheck(user, SKILL_ENGINEER, SKILL_ENGINEER_TRAINED))
 			to_chat(user, SPAN_WARNING("You don't have the training to fire [src]."))
 			return
 		if(busy)
 			to_chat(user, SPAN_WARNING("Someone else is currently using [src]."))
 			return
-		if(!is_ground_level(z))
-			to_chat(user, SPAN_WARNING("You cannot fire [src] here."))
-			return
-		if(targ_x == 0 && targ_y == 0) //Mortar wasn't set
-			to_chat(user, SPAN_WARNING("[src] needs to be aimed first."))
-			return
-		var/turf/T = locate(targ_x + dial_x + offset_x, targ_y + dial_y + offset_y, z)
-		if(!T)
-			to_chat(user, SPAN_WARNING("You cannot fire [src] to this target."))
-			return
-		var/area/A = get_area(T)
-		if(!istype(A))
-			to_chat(user, SPAN_WARNING("This area is out of bounds!"))
-			return
-		if(CEILING_IS_PROTECTED(A.ceiling, CEILING_PROTECTION_TIER_2) || protected_by_pylon(TURF_PROTECTION_MORTAR, T))
-			to_chat(user, SPAN_WARNING("You cannot hit the target. It is probably underground."))
-			return
-		if(SSticker.mode && MODE_HAS_TOGGLEABLE_FLAG(MODE_LZ_PROTECTION) && A.is_landing_zone)
-			to_chat(user, SPAN_WARNING("You cannot bomb the landing zone!"))
-			return
-
-		//Small amount of spread so that consecutive mortar shells don't all land on the same tile
-		var/turf/T1 = locate(T.x + pick(-1,0,0,1), T.y + pick(-1,0,0,1), T.z)
-		if(T1)
-			T = T1
+		if(!ship_side)
+			if(targ_x == 0 && targ_y == 0) //Mortar wasn't set
+				to_chat(user, SPAN_WARNING("[src] needs to be aimed first."))
+				return
+			if(!T)
+				to_chat(user, SPAN_WARNING("You cannot fire [src] to this target."))
+				return
+			if(!istype(A))
+				to_chat(user, SPAN_WARNING("This area is out of bounds!"))
+				return
+			if(CEILING_IS_PROTECTED(A.ceiling, CEILING_PROTECTION_TIER_2) || protected_by_pylon(TURF_PROTECTION_MORTAR, T))
+				to_chat(user, SPAN_WARNING("You cannot hit the target. It is probably underground."))
+				return
+			if(SSticker.mode && MODE_HAS_TOGGLEABLE_FLAG(MODE_LZ_PROTECTION) && A.is_landing_zone)
+				to_chat(user, SPAN_WARNING("You cannot bomb the landing zone!"))
+				return
+
+		if(!ship_side)
+			var/turf/T1 = locate(T.x + pick(-1,0,0,1), T.y + pick(-1,0,0,1), T.z) //Small amount of spread so that consecutive mortar shells don't all land on the same tile
+			if(T1)
+				T = T1
+		else
+			var/crash_occurred = (SSticker?.mode?.is_in_endgame)
+			if(crash_occurred)
+				var/turf/our_turf = get_turf(src)
+				T = our_turf
+				travel_time = 0.5 SECONDS
+			else
+				to_chat(user, SPAN_RED("ARE YOU OUT OF YOUR MIND?!"))
```

Comment:
not really a great to_chat

Reply:
pretty sure its just a copypaste from c4 code